pkg/util/pool/fast_releasing_pool.go

```diff
@@ -0,0 +1,123 @@
+package pool
+
+import (
+	"fmt"
+	"sync"
+)
+
+// FastReleasingSlabPool is similar to SlabPool, but allows for fast release of slabs if they are not used anymore.
+type FastReleasingSlabPool[T any] struct {
+	delegate Interface
+	slabSize int
+
+	mtx   sync.Mutex
+	slabs []*trackedSlab[T] // All slabs. Slab ID is an index into this slice.
+}
+
+type trackedSlab[T any] struct {
+	slab          []T
+	references    int // How many slices from this slab were returned via Get.
+	nextFreeIndex int
+}
+
+// NewFastReleasingSlabPool returns a new "fast-releasing" slab pool.
+func NewFastReleasingSlabPool[T any](delegate Interface, slabSize int) *FastReleasingSlabPool[T] {
+	return &FastReleasingSlabPool[T]{
+		delegate: delegate,
+		slabSize: slabSize,
+		slabs:    []*trackedSlab[T]{nil}, // slabId = 0 is invalid.
```

Comment:
Given the first one is always nil, does it make sense to initialise the slice with a capacity of 2? Or even an higher initial capacity to reduce the likelihood of a re-slice.